### PR TITLE
Contact us page

### LINKED ISSUE #803: FordLabs easily knows when visitors to PeopleMover attempt to contact
# Feature Request

**As a** FordLabs team member who cares about PeopleMover users
**I would** like to see a Slack message when help is needed from an end user
**so that** I can ensure their concerns and/or needs are addressed

## Context
Currently, the Contact Us link in the PeopleMover footer opens a default email client to a draft email addressed the Ford CDSID `pplmover` -- but this inbox is only accessed regularly by a small group of individuals. This 

## Acceptance Criteria

- When I click the "Contact Us" link in the PeopleMover app's footer (no matter what page I'm on), then I am taken to the Contact Us page
- When I look at the empty Contact Us form, I see that "Send" is disabled
- When I fill in only some fields on the Contact Us form, I see that "Send" remains disabled
- When I have filled in all fields on the Contact Us form, I see that "Send" has now enabled
- When I lose focus on a field after leaving it empty or with invalid email, I see a validation warning (see mockups)
- When I click "Send", I see a confirmation that my message has been sent, and the form clears
- When I log into Slack as a FordLabs team member, I see that the #peoplemover-contact channel now has a notification with the message content

## Accessibility Implemented (optional)

1. Constraint 1
2. Constraint 2
3. etc

### Mockups
image

Link to Figma: https://www.figma.com/file/S4XKeovlQqeRk64k2qzhf8gL/PeopleMover?node-id=9452%3A11173
image

#### Validation
image

image

### PR DESCRIPTION
**Pull request by RandomAshes**
## Issue
Resolves #803 

## What was done
- [x] Created a contact us page

## How to test
1. Go to any page in the app, scroll down, and click the contact us button in the footer. Ensure you're taken to the contact us page.
2. Try to submit the form without filling out all fields. Ensure you get shown a message saying to please fill out the empty fields.
3. Fill out the form fully and submit the form. Ensure that an api call to `/api/contact-us` gets made, the form clears, and the send button gets swapped out for a confirmation button and a link back to the dashboard.

### CODE REVIEW

File: api/src/main/kotlin/com/ford/internalprojects/peoplemover/contactus/ContactUsResponse.kt [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.ford.internalprojects.peoplemover.contactus
+
+class ContactUsResponse {
```
Comment (tsatam) [inline api/src/main/kotlin/com/ford/internalprojects/peoplemover/contactus/ContactUsResponse.kt]: What is this class's purpose?

File: api/src/main/resources/application-cloud.properties [hunk 2]
```diff
@@ -16,3 +16,4 @@
 #
 
 spring.flyway.ignore-missing-migrations=true
+slack.url=""
```
Comment (tsatam) [inline api/src/main/resources/application-cloud.properties:19]: If this property is set in application.properties, you don't need to include it in other profiles unless you want to change its value

File: api/src/main/resources/application-h2.properties [hunk 3]
```diff
@@ -29,3 +29,5 @@ spring.datasource.password=
 spring.datasource.driver-class-name=org.h2.Driver
 
 spring.flyway.locations=classpath:db/h2
+
+slack.url=""
```
Comment (tsatam) [inline api/src/main/resources/application-h2.properties:33]: If this property is set in application.properties, you don't need to include it in other profiles unless you want to change its value

**Review by tsatam — CHANGES_REQUESTED** [review]

File: api/src/main/kotlin/com/ford/internalprojects/peoplemover/contactus/ContactUsResponse.kt (hunk 1 above)
Comment (RandomAshes) [reply api/src/main/kotlin/com/ford/internalprojects/peoplemover/contactus/ContactUsResponse.kt]: Didn't end up using. Deleted.

**Review by tsatam — APPROVED** [review]

**Review by jwcherney — APPROVED** [review]

**Pull request merged**